providers/implementations/digests/sha3_prov.c

```diff
@@ -483,7 +488,9 @@ static void *name##_newctx(void *provctx)                                      \
                                                                                \
     if (ctx == NULL)                                                           \
         return NULL;                                                           \
-    ossl_sha3_init(ctx, pad, bitlen);                                          \
+    ossl_keccak_init(ctx, pad, bitlen, mdlen);                                 \
+    if (mdlen == SIZE_MAX)                                                     \
+        ctx->md_size = SIZE_MAX;                                               \
```

Comment:
I'm not entirely convinced that `SIZE_MAX` is the better indicator for indefinite length.  How about zero?

Reply:
`SIZE_MAX` is better IMO.  It requires less special case code in callers and generating more than `SIZE_MAX` output is likely to be highly suspect and not storable or processable in reasonable time.

Reply:
I disagree. Zero is a much better indicator that you haven't initialised something to a fixed length. 
This is *C* code after all.

Haven't we used the pattern of using 0 elsewhere - I don't know that we have used a SIZE_MAX approach to indicate indefinite elsewhere ... 

Consistency here should be the guide.

Reply:
I doubt there will be much of a difference for the caller.  Basically, this would work as an indicator whether it's appropriate to Squeeze without limit rather than using DigestFinalXOF, at least as long as we don't really have any other way for the caller to check what sort of functionality is supported by the implementation.

So in the end, it's this check:

``` C
    // Can I squeeze?
    if (md_size == SIZE_MAX)
        ...
```

or this:

``` C
    // Can I squeeze?
    if (md_size == 0)
        ...
```

Reply:
We do use `SIZE_MAX` quite a bit, but that's in cases where size maximums are a thing.  SHAKE isn't expressed in terms of maximum size, it's expressed in "indefinite" terms (even though there are admonitions about practicality).

Furthermore, `md_size` is usually *the (fixed) size*, not a maximum, and `SIZE_MAX` is a bit weird of an exception from that interpretation compare to zero, in my mind.

Reply:
I require `X` bytes of output.

Thus either:
`    if (X < max_size || max_size == 0)`
versus
`    if (X < max_size)`

The latter is, **by far**, the simpler pattern IMO.

Reply:
I'm definitely not advocating for the essentially equivalent:
`    if (X < max_size - 1)`
which is very counter intuitive.

Reply:
The problem is that `0` value is already very special and we already have a test (evp_test) that depends on `EVP_DigestFinalXOF(ctx, ptr, 0)` returning success! So no, we have to keep SIZE_MAX as the _unset_ value.